docs/FAQ.md

```diff
@@ -1,19 +1,35 @@
-# Limitations and Common Issues 
+# Frequently Asked Questions
 
-## Unity SDK
-### Headless Mode
-If you enable Headless mode, you will not be able to collect visual 
-observations from your agents.
 
-### Rendering Speed and Synchronization
-Currently the speed of the game physics can only be increased to 100x 
-real-time. The Academy also moves in time with FixedUpdate() rather than 
-Update(), so game behavior tied to frame updates may be out of sync. 
+### Scripting Runtime Environment not setup correctly
 
-## Python API
+If you hasn't switched your scripting runtime version from .NET 3.5 to .NET 4.6, you will see such error message:
 
-### Python version
-As of version 0.3, we no longer support Python 2. 
+```
+Assets/ML-Agents/Scripts/ExternalCommunicator.cs(259,24): error CS1061: Type `System.Text.StringBuilder' does not contain a definition for `Clear' and no extension method `Clear' of type `System.Text.StringBuilder' could be found. Are you missing an assembly reference?
+```
+
+This is because .NET 3.5 doesn't support method Clear() for StringBuilder, refer to [here](Installation.md#setting-up-ml-agent-within-unity) for solution. 
+
+### TensorflowSharp flag not turned on. 
+
+If you have already imported the TensorflowSharp plugin, but havn't set ENABLE_TENSORFLOW flag for your scripting define symbols, you will see the following error message:
+
+```
+You need to install and enable the TensorflowSharp plugin inorder to use the internal brain. 
+```
+
+This error message occurs because the TensorflowSharp plugin won't be usage without the ENABLE_TENSORFLOW flag, refer to [here](Installation.md#setting-up-ml-agent-within-unity) for solution. 
+
+### Tensorflow variable name error
+
+If you have some variables set in Unity and the Tensorflow side doesn't know about them, you will see some error like this:
```

Comment:
This is a little vague. TensorFlow shouldn't know about most of your Unity variables!. We need to be more specific. Similarly, what exactly the "TensorFlow side" knows or doesn't know is unclear.